bimsb/packages/bioinformatics-nonfree.scm

```diff
@@ -1882,3 +1883,72 @@ MACE is a bioinformatics tool dedicated to analyze ChIP-exo data.")
     ;; others say "free for all use", others say "GPL version 1.3" (?), yet
     ;; others say "Artistic license".
     (license license:gpl2+)))
+
+;; Although this program is released under the GPL it depends on
+;; ViennaRNA and bowtie1, which are non-free software.
+(define-public mirdeep2
+  (package
+    (name "mirdeep2")
+    (version "0.0.92")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                   (url "https://github.com/rajewsky-lab/mirdeep2.git")
+                   (commit (string-append "v" version))))
+              (file-name (string-append name "-" version "-checkout"))
+              (sha256
+               (base32
+                "1gasx6vb38n4vbjbn5s7ch2rpflgf2crdd8mdp2xm571w2j01wmm"))))
+    (build-system gnu-build-system)
+    (arguments
+     `(#:phases
+       (modify-phases %standard-phases
+         (delete 'configure)
+         (delete 'build)
+         (delete 'check)
+         (replace 'install
+           (lambda* (#:key inputs outputs #:allow-other-keys)
+             (let ((target (string-append (assoc-ref outputs "out")
+                                          "/bin")))
+               (with-directory-excursion "src"
+                 (copy-recursively "." target))
+               (copy-file "Rfam_for_miRDeep.fa"
+                          (string-append  target "/../Rfam_for_miRDeep.fa"))
+               (with-output-to-file (string-append target "/../install_successful")
+                 (const #t)))  ;; simply touch the file
+             #t))
+         (add-after 'install 'wrap-perl-scripts
+          (lambda* (#:key inputs outputs #:allow-other-keys)
+            (let ((out (assoc-ref outputs "out")))
+              ;; Make sure perl scripts find all perl inputs at runtime.
+              (for-each (lambda (prog)
+                          (wrap-program (string-append out "/bin/" prog)
+                            `("PERL5LIB" ":" prefix
+                              (,(getenv "PERL5LIB")))))
+                        '("make_html2.pl"
+                          "make_html.pl"
+                          "miRDeep2.pl"))
+              ;; Make sure perl scripts find all input binaries at runtime.
+              (for-each (lambda (prog)
+                          (wrap-program (string-append out "/bin/" prog)
+                            `("PATH" ":" prefix
+                               (,(getenv "PATH")))))
+                        '("make_html2.pl"
+                          "make_html.pl"
+                          "mapper.pl"
+                          "miRDeep2_core_algorithm.pl"
+                          "miRDeep2.pl"
+                          "prepare_signature.pl"
+                          "quantifier.pl"))
+              #t))))))
+    (inputs
+      `(("bowtie1" ,bowtie1)
+       ("perl-pdf-api2" ,perl-pdf-api2)
+       ("perl" ,perl)
+       ("randfold" ,randfold)
+       ("viennarna" ,viennarna-2.2.10)))
```

Comment:
I still would prefer to depend on `viennarna` as in 'latest' here to benefit from updates.
There are however several versions assigned to variables and used as inputs with comments regarding why they depend on a specific version. So I did not dare to touch this mess. :wink:

Reply:
Yeah, I should rebind these variables so that `viennarna` is actually bound to the latest version.  Will do this after merging this patch.

Reply:
@rekado: This is just a reminder to rebind `viennarna` after merging this PR. :wink: